### PR TITLE
Don't require ticket status and type

### PR DESCRIPTION
**Pull request by laurenleach**
- We don't require status but if the schema didn't include status we were still erroring 
- Change ticket type to not be required since not all ticket systems require type

### CODE REVIEW

File: pkg/types/ticket/custom_fields.go [hunk 1]
```diff
@@ -265,10 +265,11 @@ func ValidateTicket(ctx context.Context, schema *v2.TicketSchema, ticket *v2.Tic
 	}
 
 	// Look for a matching ticket type
-	foundMatch = false
+	foundMatch = ticket.Type == nil
 	for _, tType := range schema.GetTypes() {
 		if ticket.Type == nil {
```
Comment (mgaeta) [inline pkg/types/ticket/custom_fields.go:270]: The code block
```go
if ticket.Type == nil {
	foundMatch = true
	break
}
```
can be removed from the loop:
```go
// Look for a matching status
// Status is not required
foundMatch := ticket.Status == nil ||
if !foundMatch {
  for _, status := range schema.GetStatuses() {
    if ticket.Status.GetId() == status.GetId() {
	    foundMatch = true
	    break
    }
  }
}
```

File: pkg/types/ticket/custom_fields.go [hunk 2]
```diff
@@ -265,10 +265,11 @@ func ValidateTicket(ctx context.Context, schema *v2.TicketSchema, ticket *v2.Tic
 	}
 
 	// Look for a matching ticket type
-	foundMatch = false
+	foundMatch = ticket.Type == nil
```
Comment (btipling) [inline pkg/types/ticket/custom_fields.go:268]: I might add a more specific comment here and above to explain why we do this, if a ticket doesn't have a status or a type we don't need to validate, skip the loop in this case

File: pkg/types/ticket/custom_fields.go [hunk 3]
```diff
@@ -265,10 +265,11 @@ func ValidateTicket(ctx context.Context, schema *v2.TicketSchema, ticket *v2.Tic
 	}
 
 	// Look for a matching ticket type
-	foundMatch = false
```
Comment (btipling) [inline pkg/types/ticket/custom_fields.go:268]: `hasValidTicketType`

File: pkg/types/ticket/custom_fields.go [hunk 4]
```diff
@@ -245,7 +245,7 @@ func ValidateTicket(ctx context.Context, schema *v2.TicketSchema, ticket *v2.Tic
 	l := ctxzap.Extract(ctx)
 
 	// Look for a matching status
-	foundMatch := false
```
Comment (btipling) [inline pkg/types/ticket/custom_fields.go:248]: `validTicketStatus`

**Review by btipling — APPROVED** [review]

**Pull request merged**